### PR TITLE
Adds support for specifying a role path in watches file.

### LINKED ISSUE #20: improvements to config file
These improvements to `/opt/ansible/config.yaml` were suggested in a meeting this week. Here is how the format currently looks:

```
- name: Database
  version: v1alpha1
  group: app.example.com
  kind: Database
  path: /opt/ansible/roles/busybox/playbook.yaml
```

`path` currently only can reference a playbook. Since we want to support running either a playbook or a role directly, the `path` key will be replaced by `playbook` and `role` keys. They will be mutually exclusive.

The `name` key was deemed to not be valuable, so it should be removed.

### PR DESCRIPTION
**Pull request by mhrivnak**
This required making the changes in #20 in addition to minor plumbing.

fixes #20

### CODE REVIEW

File: pkg/runner/runner.go [hunk 1]
```diff
@@ -57,7 +57,20 @@ func NewFromWatches(path string) (map[schema.GroupVersionKind]Runner, error) {
 			Version: w.Version,
 			Kind:    w.Kind,
 		}
-		m[s] = NewForPlaybook(w.Path, s)
+		switch {
+		case w.Playbook != "":
+			if !filepath.IsAbs(w.Playbook) {
```
[the remark this answers is not in the document]
Comment (mhrivnak) [reply pkg/runner/runner.go]: If we accept a relative path, then we have to document what it should be relative to. I just don't see a use case there, but if there is, we can certainly do it.

File: README.md [hunk 2]
```diff
@@ -148,18 +148,40 @@ COPY watches.yaml /opt/ansible/watches.yaml
...
+
+Example specifying a role:
+
+```yaml
+---
+- version: v1alpha1
   group: app.example.com
   kind: Database
-  path: /opt/ansible/roles/busybox/playbook.yaml
+  playbook: /opt/ansible/roles/busybox/
```
Comment (shawn-hurley) [inline README.md]: Shouldn't this be `role`?

File: pkg/runner/runner.go [hunk 3]
```diff
@@ -125,6 +138,15 @@ func (r *runner) Run(u *unstructured.Unstructured, kubeconfig string) (chan even
 			"runner_http_path": receiver.URLPath,
 		},
 	}
+	// If Path is a dir, assume it is a role path. Otherwise assume it's a
```
[the remark this answers is not in the document]
Comment (fabianvf) [reply pkg/runner/runner.go]: going to ping @jimi-c on that to be sure, but as far as I understand it a role can't be a file and a playbook can't be a directory. 

Out of curiousity, if that's true would it be better for the user to continue just passing a path and we can use this same logic for figuring out what to do?

[the remark this answers is not in the document]
Comment (mhrivnak) [reply pkg/runner/runner.go]: Fair question. I think it's better to be explicit about what the user is intending to provide. The IsDir approach seems sufficient for internal purposes to distinguish which we're working with at any given moment, but for a user interface, auto-detection of what the user provided can turn into a minefield.

[the remark this answers is not in the document]
Comment (fabianvf) [reply pkg/runner/runner.go]: +1

**Pull request merged**